StringOperations.java

```diff
@@ -0,0 +1,146 @@
+/***
+ * Program to perform String operations such as Append, CountWords, Replace, isPalindrome and Splice.
+ * Owner: Uddeshya Patidar
+ * Date: 04/09/2024
+ */
+package practice.java;
+
+import java.util.Scanner;
+
+public class StringOperations {
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+
+        do {
+            System.out.print("Enter a string: ");
+            String inputString = scanner.nextLine();
+            System.out.println("\nChoose an operation:"
+            		+ "\n1. Append"
+            		+ "\n2. Count Words"
+            		+ "\n3. Replace"
+            		+ "\n4. Is Palindrome"
+            		+ "\n5. Splice");
+            
+            System.out.print("Enter the number of the operation: ");
+            
+            // Check input
+            if (!scanner.hasNextInt()) {
+                System.out.println("Invalid input: Please enter a valid operation number.");
+                scanner.next(); 
+                continue; 
+            }
+            int choice = scanner.nextInt();
+            scanner.nextLine(); // Consume the newline character
+
+            String resultString = inputString;
+            boolean validOption = true;
+
+            switch (choice) {
+                case 1:// Append
+                    System.out.print("Enter the string to append: ");
+                    String appendString = scanner.nextLine();
+                    if (!appendString.isEmpty()) {
+                        resultString = inputString + appendString;
+                        System.out.println("Updated String: " + resultString);
+                    } else {
+                        System.out.println("Invalid input: The string to append cannot be empty.");
+                    }
+                    break;
+
+                case 2:// Count words 
+                    int wordCount = 0;
+                    boolean inWord = false;
+                    for (int i = 0; i < inputString.length(); i++) {
+                        if (inputString.charAt(i) != ' ') {
+                            if (!inWord) {
+                                wordCount++;
+                                inWord = true;
+                            }
+                        } else {
+                            inWord = false;
+                        }
+                    }
+                    System.out.println("Word count: " + wordCount);
+                    break;
+
+                case 3:// replace 
+                    System.out.print("Enter the character or substring to replace: ");
```

Comment:
The nested loops in the replace operation are overly complex. Consider using helper functions or splitting the logic for readability.
 Align variable declarations like StringBuilder replaceResult = new StringBuilder(); with other lines for visual consistency.